File: daq/slc/nsm/src/NSMCommand.cc
/**************************************************************************
 * basf2 (Belle II Analysis Software Framework)                           *
 * Author: The Belle II Collaboration                                     *
 *                                                                        *
 * See git log for contributors and copyright holders.                    *
 * This file is licensed under LGPL-3.0, see LICENSE.md.                  *
 **************************************************************************/
#include "daq/slc/nsm/NSMCommand.h"

using namespace Belle2;

const NSMCommand NSMCommand::OK(1, "OK");
const NSMCommand NSMCommand::ERROR(2, "ERROR");
const NSMCommand NSMCommand::FATAL(3, "FATAL");
const NSMCommand NSMCommand::STATE(4, "STATE");
const NSMCommand NSMCommand::LOG(5, "LOG");
const NSMCommand NSMCommand::LOGSET(6, "LOGSET");
const NSMCommand NSMCommand::LOGGET(7, "LOGGET");
const NSMCommand NSMCommand::VGET(11, "VGET");
const NSMCommand NSMCommand::VSET(12, "VSET");
const NSMCommand NSMCommand::VREPLY(13, "VREPLY");
const NSMCommand NSMCommand::VLISTGET(14, "VLISTGET");
const NSMCommand NSMCommand::VLISTSET(15, "VLISTSET");
const NSMCommand NSMCommand::DATAGET(401, "DATAGET");
const NSMCommand NSMCommand::DATASET(402, "DATASET");
const NSMCommand NSMCommand::DBSET(403, "DBSET");
const NSMCommand NSMCommand::DBGET(404, "DBGET");
const NSMCommand NSMCommand::DBLISTSET(405, "DBLISTSET");
const NSMCommand NSMCommand::DBLISTGET(406, "DBLISTGET");
const NSMCommand NSMCommand::LOGLIST(407, "LOGLIST");

const NSMCommand& NSMCommand::operator=(const std::string& label)
{
  if (label == LOG.getLabel()) *this = LOG;
  else if (label == OK.getLabel()) *this = OK;
  else if (label == ERROR.getLabel()) *this = ERROR;
  else if (label == FATAL.getLabel()) *this = FATAL;
  else if (label == LOG.getLabel()) *this = LOG;
  else if (label == LOGSET.getLabel()) *this = LOGSET;
  else if (label == LOGGET.getLabel()) *this = LOGGET;
  else if (label == STATE.getLabel()) *this = STATE;
  else if (label == VGET.getLabel()) *this = VGET;
  else if (label == VSET.getLabel()) *this = VSET;
  else if (label == VREPLY.getLabel()) *this = VREPLY;
  else if (label == VLISTGET.getLabel()) *this = VLISTGET;
  else if (label == VLISTSET.getLabel()) *this = VLISTSET;
  else if (label == DATAGET.getLabel()) *this = DATAGET;
  else if (label == DATASET.getLabel()) *this = DATASET;
  else if (label == DBSET.getLabel()) *this = DBSET;
  else if (label == DBGET.getLabel()) *this = DBGET;
  else if (label == DBLISTSET.getLabel()) *this = DBLISTSET;
  else if (label == DBLISTGET.getLabel()) *this = DBLISTGET;
  else if (label == LOGLIST.getLabel()) *this = LOGLIST;
  else *this = Enum::UNKNOWN;
  return *this;
}

const NSMCommand& NSMCommand::operator=(int id)
{
  if (id == LOG.getId()) *this = LOG;
  else if (id == OK.getId()) *this = OK;
  else if (id == ERROR.getId()) *this = ERROR;
  else if (id == FATAL.getId()) *this = FATAL;
  else if (id == LOG.getId()) *this = LOG;
  else if (id == LOGSET.getId()) *this = LOGSET;
  else if (id == LOGGET.getId()) *this = LOGGET;
  else if (id == STATE.getId()) *this = STATE;
  else if (id == VSET.getId()) *this = VSET;
  else if (id == VGET.getId()) *this = VGET;
  else if (id == VREPLY.getId()) *this = VREPLY;
  else if (id == VLISTSET.getId()) *this = VLISTSET;
  else if (id == VLISTGET.getId()) *this = VLISTGET;
  else if (id == DATASET.getId()) *this = DATASET;
  else if (id == DATAGET.getId()) *this = DATAGET;
  else if (id == DBSET.getId()) *this = DBSET;
  else if (id == DBGET.getId()) *this = DBGET;
  else if (id == DBLISTSET.getId()) *this = DBLISTSET;
  else if (id == DBLISTGET.getId()) *this = DBLISTGET;
  else if (id == LOGLIST.getId()) *this = LOGLIST;
  else *this = Enum::UNKNOWN;
  return *this;
}

const NSMCommand& NSMCommand::operator=(const char* label)
{
  if (label != NULL)  *this = std::string(label);
  else *this = Enum::UNKNOWN;
  return *this;
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/nsm/NSMCommand.java
package b2daq.nsm;

import b2daq.core.Enum;

public class NSMCommand extends Enum {

    final static public int DISABLED = 0;
    final static public int ENABLED = 1;
    final static public int SUGGESTED = 2;

    final static public NSMCommand UNKNOWN = new NSMCommand(0, "UNKNOWN");
    final static public NSMCommand OK = new NSMCommand(1, "OK");
    final static public NSMCommand ERROR = new NSMCommand(2, "ERROR");
    final static public NSMCommand LOG = new NSMCommand(5, "LOG");
    final static public NSMCommand VGET = new NSMCommand(11, "VGET");
    final static public NSMCommand VSET = new NSMCommand(12, "VSET");
    final static public NSMCommand VLISTGET = new NSMCommand(13, "VLISTGET");
    final static public NSMCommand VLISTSET = new NSMCommand(14, "VLISTSET");
    final static public NSMCommand VREPLY = new NSMCommand(15, "VREPLY");
    final static public NSMCommand DATASET = new NSMCommand(401, "DATASET");
    final static public NSMCommand DATAGET = new NSMCommand(402, "DATAGET");
    final static public NSMCommand DBSET = new NSMCommand(403, "DBSET");
    final static public NSMCommand DBGET = new NSMCommand(404, "DBGET");
    final static public NSMCommand DBLISTSET = new NSMCommand(405, "DBLISTSET");
    final static public NSMCommand DBLISTGET = new NSMCommand(406, "DBLISTGET");
    final static public NSMCommand LOGLIST = new NSMCommand(407, "LOGLIST");

    public NSMCommand() {
        copy(NSMCommand.UNKNOWN);
    }

    protected NSMCommand(int id, String label) {
        super(id, label);
    }

    public NSMCommand(NSMCommand command) {
        copy(command);
    }

    public NSMCommand(String command) {
        copy(command);
    }

    public boolean copy(String msg) {
        if (msg.equals(UNKNOWN.getLabel())) {
            copy(UNKNOWN);
            return true;
        } else if (msg.equals(OK.getLabel())) {
            copy(OK);
            return true;
        } else if (msg.equals(ERROR.getLabel())) {
            copy(ERROR);
            return true;
        } else if (msg.equals(LOG.getLabel())) {
            copy(LOG);
            return true;
        } else if (msg.equals(VGET.getLabel())) {
            copy(VGET);
            return true;
        } else if (msg.equals(VSET.getLabel())) {
            copy(VSET);
            return true;
        } else if (msg.equals(VLISTGET.getLabel())) {
            copy(VLISTGET);
            return true;
        } else if (msg.equals(VLISTSET.getLabel())) {
            copy(VLISTSET);
            return true;
        } else if (msg.equals(VREPLY.getLabel())) {
            copy(VREPLY);
            return true;
        } else if (msg.equals(DATAGET.getLabel())) {
            copy(DATAGET);
            return true;
        } else if (msg.equals(DATASET.getLabel())) {
            copy(DATASET);
            return true;
        } else if (msg.equals(DBGET.getLabel())) {
            copy(DBGET);
            return true;
        } else if (msg.equals(DBSET.getLabel())) {
            copy(DBSET);
            return true;
        } else if (msg.equals(DBLISTGET.getLabel())) {
            copy(DBLISTGET);
            return true;
        } else if (msg.equals(DBLISTSET.getLabel())) {
            copy(DBLISTSET);
            return true;
        } else if (msg.equals(LOGLIST.getLabel())) {
            copy(LOGLIST);
            return true;
        }
        return false;
    }

    public boolean copy(int id) {
        if (id == UNKNOWN.getId()) {
            copy(UNKNOWN);
            return true;
        } else if (id == OK.getId()) {
            copy(OK);
            return true;
        } else if (id == ERROR.getId()) {
            copy(ERROR);
            return true;
        } else if (id == LOG.getId()) {
            copy(LOG);
            return true;
        } else if (id == VGET.getId()) {
            copy(VGET);
            return true;
        } else if (id == VSET.getId()) {
            copy(VSET);
            return true;
        } else if (id == VLISTGET.getId()) {
            copy(VLISTGET);
            return true;
        } else if (id == VLISTSET.getId()) {
            copy(VLISTSET);
            return true;
        } else if (id == VREPLY.getId()) {
            copy(VREPLY);
            return true;
        } else if (id == DATAGET.getId()) {
            copy(DATAGET);
            return true;
        } else if (id == DATASET.getId()) {
            copy(DATASET);
            return true;
        } else if (id == DBGET.getId()) {
            copy(DBGET);
            return true;
        } else if (id == DBSET.getId()) {
            copy(DBSET);
            return true;
        } else if (id == DBLISTGET.getId()) {
            copy(DBLISTGET);
            return true;
        } else if (id == DBLISTSET.getId()) {
            copy(DBLISTSET);
            return true;
        } else if (id == LOGLIST.getId()) {
            copy(LOGLIST);
            return true;
        }
        return false;
    }

}
